Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -26,70 +37,79 @@ export default function widgetBehavior(publicAPI, model) {
   // Interactor event
   // --------------------------------------------------------------------------
 
+  model.handleVisibility = !(
+    !model.handle1Visibility && !model.handle2Visibility
+  );
+
   function ignoreKey(e) {
     return e.altKey || e.controlKey || e.shiftKey;
   }
 
+  function updateCursor() {
+    model.widgetState.setIsDragging(true);
+    model.openGLRenderWindow.setCursor('grabbing');
+    model.interactor.requestAnimation(publicAPI);
+  }
+
+  // --------------------------------------------------------------------------
+  // Text Methods
+  // --------------------------------------------------------------------------
+
   /*
+   * DetectOffsetDirectionForTextPosition
    * check for handle 2 position in comparison to handle 1 position
    * and sets text offset to not overlap on the line representation
    */
-  function calcTextPosWithLineAngle() {
+
+  function detectOffsetDirectionForTextPosition() {
     const pos1 = model.widgetState.getHandle1().getOrigin();
     const pos2 = model.widgetState.getHandle2().getOrigin();
-    const SVGTextProps = model.representations[2].getTextProps();
 
     let dySign = 1;
     if (pos1[0] <= pos2[0]) {
       dySign = pos1[1] <= pos2[1] ? 1 : -1;
     } else {
       dySign = pos1[1] <= pos2[1] ? -1 : 1;
     }
-    SVGTextProps.dy = dySign * Math.abs(SVGTextProps.dy);
+    return dySign;
   }
 
-  function updateHandleDirection(behavior, callData) {
-    let bv = behavior;
-    if (bv === HandleBehavior.HANDLE1_ALONE) {
-      const handle1Pos = model.widgetState.getHandle1().getOrigin();
-      const WorldMousePos = publicAPI.computeWorldToDisplay(
-        model.renderer,
-        handle1Pos[0],
-        handle1Pos[1],
-        handle1Pos[2]
-      );
-      const mousePos = publicAPI.computeDisplayToWorld(
-        model.renderer,
-        callData.position.x,
-        callData.position.y,
-        WorldMousePos[2]
-      );
-      vtkMath.subtract(
-        model.widgetState.getHandle1().getOrigin(),
-        mousePos,
-        Direction
-      );
-      bv = 0;
-    } else {
-      const modifier = bv === 1 ? 1 : -1;
-      bv -= 1;
-      const handle1Pos = model.widgetState.getHandle1().getOrigin();
-      const handle2Pos = model.widgetState.getHandle2().getOrigin();
-      vtkMath.subtract(handle1Pos, handle2Pos, Direction);
-      vtkMath.multiplyScalar(Direction, modifier);
-    }
-    model.representations[bv].getGlyph().setDirection(Direction);
+  function calcTextPosWithLineAngle() {
```

Comment:
calcTextPosWithLineAngle -> updateTextPosition()

+ add documentation on this function to explain what's going on